infoschema/builder.go

```diff
@@ -367,6 +369,38 @@ func (b *Builder) applyDropSchema(schemaID int64) []int64 {
 	return tableIDs
 }
 
+func (b *Builder) applyRecoverSchema(m *meta.Meta, diff *model.SchemaDiff) error {
+	if di, ok := b.is.SchemaByID(diff.SchemaID); ok {
+		return ErrDatabaseExists.GenWithStackByArgs(
+			fmt.Sprintf("(Schema ID %d)", di.ID),
+		)
+	}
+	di, err := m.GetDatabase(diff.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err = b.applyPlacementUpdate(placement.GroupID(diff.SchemaID)); err != nil {
+		return errors.Trace(err)
+	}
```

Comment:
cc @djshow832 

I'm not sure whether the placement rule also works for `database`, I thought placement rule was only use for `table`.